spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1331,6 +1331,44 @@ The connector is responsible for the acknowledgment (positive or negative) of th
 * An outgoing connector must acknowledge the incoming `org.eclipse.microprofile.reactive.messaging.Message` once it has successfully dispatched the message.
 * An outgoing connector must acknowledge negatively the incoming `org.eclipse.microprofile.reactive.messaging.Message` if it cannot be dispatched.
 
+== Health
+
+When MicroProfile Reactive Messaging is used in an environment where MicroProfile Health is enabled, implicit readiness
+and liveness checks `messaging` are produced.
+
+[source, json]
+----
+{
+  "status": "UP",
+  "checks": [
+    {
+      "name": "messaging",
+      "status": "UP"
+    }
+  ]
+}
+----
+
+=== Readiness
+State of the readiness check `messaging` is DOWN until subscribers of all messaging channels are subscribed to their
+respective publishers and `onSubscribe` signal is detected on all channels.
+
+=== Liveness
+
+State of the liveness check `messaging` is UP until `cancel` or `onError` signals are detected on any of the messaging
+channels.
+
+=== Excluding channels
+
+For channels which are not desired to be checked for liveness or readiness, exclusion is configurable with
+`mp.messaging.health.live.exclude` or `mp.messaging.health.ready.exclude` config keys.
+
+[source, properties]
+----
+mp.messaging.health.live.exclude=<channel-name>,<channel-name>
+mp.messaging.health.ready.exclude=<channel-name>,<channel-name>
```

Comment:
If we follow the logic I described above, this would only make sense for subscriber methods. For connectors, it would be better to have a specific attribute and keep the connector configuration co-localized.